internal/controllers/serviceinstance_controller.go

```diff
@@ -206,34 +208,48 @@ func (r *ServiceInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			if err != nil {
 				return ctrl.Result{}, err
 			}
-
-			//Add parameters to adopt the orphaned instance
-			var parameterObjects []map[string]interface{}
-			paramMap := make(map[string]interface{})
-			paramMap["parameter-hash"] = cfinstance.ParameterHash
-			paramMap["owner"] = cfinstance.Owner
-			parameterObjects = append(parameterObjects, paramMap)
-			parameters, err := mergeObjects(parameterObjects...)
-			if err != nil {
-				return ctrl.Result{}, errors.Wrap(err, "failed to unmarshal/merge parameters")
-			}
-			// update the orphaned cloud foundry instance
-			log.V(1).Info("Updating instance")
-			if err := client.UpdateInstance(
-				ctx,
-				cfinstance.Guid,
-				spec.Name,
-				"",
-				parameters,
-				nil,
-				serviceInstance.Generation,
-			); err != nil {
-				return ctrl.Result{}, err
+			if cfinstance != nil && cfinstance.State == facade.InstanceStateReady {
```

Comment:
Before, cfinstance was not checked for nil, probably because GetInstance(..) call above is able to return (nil, nil) if it did not find an instance?

Reply:
yeah Right, when the 'GetInstance' for adopt instance retuned nil, operator went into crash as next steps depends on cf object. Also adopt instance should work if the orphan cf instance is ready not in other states, so added the condition.

Reply:
Ok